### ISSUE TITLE
Image data has type int64 fails QC report generation

### ISSUE THREAD

**Issue by sandrinebedard** [spinalcordtoolbox/spinalcordtoolbox#4420]
### Description
I want to generate a spinal canal mask, and what I do is run `sct_propseg` with the flag `- CSF`
and sum the spinal cord mask and CSF mask to get the spinal canal. After, I run the qc report to validate the mask, before sct 6.2 it worked, but now the `sct_qc` command fails since the dtype is of int64.

I can share the data on slack (cannot share publicly).

### Steps to Reproduce

0. Install SCT <release or git revision and platform, you can use the
   first lines of output of sct_check_dependencies>

## 1. Run `sct_propseg` with `-CSF`
```python
sct_propseg -i sub-DMAim1Test003_ses-spinalcord21Ch_task-tens_run-1_bold_mean.nii.gz -c t2 -CSF -qc ./qc
```
**dtype**: 
* `sub-DMAim1Test003_ses-spinalcord21Ch_task-tens_run-1_bold_mean_seg.nii.gz` : INT8
* `sub-DMAim1Test003_ses-spinalcord21Ch_task-tens_run-1_bold_mean_CSF_seg.nii.gz`: UINT8

## 2. Sum spinal cord + CSF
```python
sct_maths -i sub-DMAim1Test003_ses-spinalcord21Ch_task-tens_run-1_bold_mean_seg.nii.gz -add sub-DMAim1Test003_ses-spinalcord21Ch_task-tens_run-1_bold_mean_CSF_seg.nii.gz -o sub-DMAim1Test003_ses-spinalcord21Ch_task-tens_run-1_bold_mean_CANAL_seg.nii.gz
```
**dtype**: 
* `sub-DMAim1Test003_ses-spinalcord21Ch_task-tens_run-1_bold_mean_CANAL_seg.nii.gz`: INT64

So `sct_maths` changes the dtype to INT64, if I add `-type uint8` it outputs the right type. I don't know if the default type changed?

It is not useful if it is not compatible with the QC

## 3. Run QC
```python
sct_qc -i sub-DMAim1Test003_ses-spinalcord21Ch_task-tens_run-1_bold_mean.nii.gz -s sub-DMAim1Test003_ses-spinalcord21Ch_task-tens_run-1_bold_mean_CANAL_seg.nii.gz -p sct_deepseg_sc -qc ./qc
```
**Actual behavior:** 

```python
(base) sabeda@DESKTOP-JQ8A4UV:/mnt/c/Users/sb199/Projet3_data/dermatomal_mapping/data_processed_results_v1/data_processed/sub-DMAim1Test003/ses-spinalcord21Ch/func/run-1$ sct_qc -i sub-DMAim1Test003_ses-spinalcord21Ch_task-tens_run-1_bold_mean.nii.gz -s sub-DMAim1Test003_ses-spinalcord21Ch_task-tens_run-1_bold_mean_CANAL_seg.nii.gz -p sct_deepseg_sc -qc ./qc
/mnt/c/Users/sb199/spinalcordtoolbox/python/envs/venv_sct/lib/python3.9/site-packages/numpy/core/getlimits.py:542: UserWarning: Signature b'\x00\xd0\xcc\xcc\xcc\xcc\xcc\xcc\xfb\xbf\x00\x00\x00\x00\x00\x00' for <class 'numpy.longdouble'> does not match any known type: falling back to type probe function.
This warnings indicates broken support for the dtype!
  machar = _get_machar(dtype)

--
Spinal Cord Toolbox (git-master-9a7ecad0d3ac019f0f84ff239685a84381195daa*)

sct_qc -i sub-DMAim1Test003_ses-spinalcord21Ch_task-tens_run-1_bold_mean.nii.gz -s sub-DMAim1Test003_ses-spinalcord21Ch_task-tens_run-1_bold_mean_CANAL_seg.nii.gz -p sct_deepseg_sc -qc ./qc
--

*** Generate Quality Control (QC) html report ***
Resample images to 0.6x0.6 mm
Traceback (most recent call last):
  File "/mnt/c/Users/sb199/spinalcordtoolbox/spinalcordtoolbox/scripts/sct_qc.py", line 114, in <module>
    main(sys.argv[1:])
  File "/mnt/c/Users/sb199/spinalcordtoolbox/spinalcordtoolbox/scripts/sct_qc.py", line 98, in main
    generate_qc(fname_in1=arguments.i,
  File "/mnt/c/Users/sb199/spinalcordtoolbox/spinalcordtoolbox/reports/qc.py", line 682, in generate_qc
    qcslice = Axial([Image(fname_in1), Image(fname_seg)])
  File "/mnt/c/Users/sb199/spinalcordtoolbox/spinalcordtoolbox/reports/slice.py", line 61, in __init__
    img_r = self._resample_slicewise(img, p_resample, type_img=type_img, image_ref=image_ref)
  File "/mnt/c/Users/sb199/spinalcordtoolbox/spinalcordtoolbox/reports/slice.py", line 271, in _resample_slicewise
    nii = Nifti1Image(image.data, image.hdr.get_best_affine())
  File "/mnt/c/Users/sb199/spinalcordtoolbox/python/envs/venv_sct/lib/python3.9/site-packages/nibabel/nifti1.py", line 1836, in __init__
    alert_future_error(
  File "/mnt/c/Users/sb199/spinalcordtoolbox/python/envs/venv_sct/lib/python3.9/site-packages/nibabel/deprecated.py", line 120, in alert_future_error
    raise error_class(f'{msg} {error_rec}'.strip())
ValueError: Image data has type int64, which may cause incompatibilities with other tools. To use this type, pass an explicit header or dtype argument to Nifti1Image().
```

**Comment by joshuacwnewton** [comment]
I think this is a duplicate of:

- https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/4408

I thought I had [mostly fixed this issue](https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/4408#issuecomment-2007390996)? 

> From my investigation in #4332, since this error only occurs when `nib.Nifti1Image` is called without the `header` or `dtype` arguments, we don't have to worry about this within `Image`, since `Image` _always_ specifies `header`.
> 
> Instead, we only have to worry about this for calls to `nib.Nifti1Image` without the `header` argument. Meaning, this issue will probably be solved when we address [#3232 (comment)](https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/3232#issuecomment-1917645212), since it's meant to replace all instances of `nib.Nifti1Image` + `nib.save()` with `Image` and `Image.save()`. And, that issue was on my TODO list anyway.

I ended up not replacing instances of `Nifti1Image` -- as per your traceback, we can see the following usage:

```
  File "/mnt/c/Users/sb199/spinalcordtoolbox/spinalcordtoolbox/reports/slice.py", line 271, in _resample_slicewise
    nii = Nifti1Image(image.data, image.hdr.get_best_affine())
```

Here, we create a Nifti1Image using the data array + affine only (no header), and the error is thrown.

**Comment by joshuacwnewton** [comment]
Closing this issue as a duplicate of #4332.

I will address #4332 by ensuring that all instances of `Nifti1Image` specify the header where possible, which will unlock QC report generation.
